pkg/k8s/client.go

```diff
@@ -33,51 +32,40 @@ type KubeClient struct {
 
 // NewKubeClient new kubernetes api client
 func NewKubeClient() (Client, error) {
-
 	config, err := buildConfig()
 	if err != nil {
 		return nil, err
 	}
-
 	clientset, err := getkubeclient(config)
 	if err != nil {
 		return nil, err
 	}
-
 	crdclient, err := crd.NewCRDClientLite(config)
 	if err != nil {
 		return nil, err
 	}
-
 	kubeClient := &KubeClient{ClientSet: clientset, CrdClient: crdclient}
 
 	return kubeClient, nil
 }
 
 // GetPodName get pod ns,name from apiserver
 func (c *KubeClient) GetPodName(podip string) (podns, poddname string, err error) {
-	if c == nil {
-		return "", "", fmt.Errorf("kubeclinet is nil")
-	}
-
 	if podip == "" {
 		return "", "", fmt.Errorf("podip is empty")
 	}
-
-	//podipFieldSel := fmt.Sprintf("status.podIp=%s", podip)
-	podList, err := c.ClientSet.CoreV1().Pods("").List(metav1.ListOptions{})
+	podList, err := c.ClientSet.CoreV1().Pods("").List(metav1.ListOptions{
+		FieldSelector: "status.podIP==" + podip + ",status.phase==Running",
```

Comment:
I think this should be filtering on pods in `Pending` and `Running` state.

Reply:
The pods in pending state would not have an ip, so it should just be running state